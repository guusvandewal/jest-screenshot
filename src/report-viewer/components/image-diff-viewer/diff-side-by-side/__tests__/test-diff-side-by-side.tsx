import * as React from "react";
import { shallow } from "enzyme";
import { DiffSideBySide } from "..";

describe("DiffSideBySide", () => {
    const someProps = {
        received: "reports/something/received.png",
        snapshot: "reports/something/snapshot.png",
        diff: "reports/something/diff.png",
        width: 100,
        height: 100,
    };

    it("looks as expected", () => {
        expect(shallow(<DiffSideBySide {...someProps} />)).toMatchSnapshot();
    });

    [0, 1, 0.7].forEach(opacity => {
        it(`when setting the diff opacity to ${opacity}`, () => {
            const element = shallow(<DiffSideBySide {...someProps} />);
            const input = element.find("input[type='range']");
            input.simulate("change", { currentTarget: { value: opacity }});
            expect(element).toMatchSnapshot();
            [
                element.find("div.viewer-container div.viewer-received div.viewer-diff img"),
                element.find("div.viewer-container div.viewer-snapshot div.viewer-diff img"),
            ].forEach(viewerDiffImg => {
                expect(viewerDiffImg.prop("style").opacity).toBe(opacity);
            });
        });
    });
});
