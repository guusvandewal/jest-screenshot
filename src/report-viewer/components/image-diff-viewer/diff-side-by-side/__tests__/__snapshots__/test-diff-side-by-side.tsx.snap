// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiffSideBySide looks as expected 1`] = `
<Fragment>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0.5}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
  >
    <div
      className="viewer-received"
    >
      <img
        src="reports/something/received.png"
      />
      <div
        className="viewer-diff"
      >
        <img
          src="reports/something/diff.png"
          style={
            Object {
              "opacity": 0.5,
            }
          }
        />
      </div>
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
      <div
        className="viewer-diff"
      >
        <img
          src="reports/something/diff.png"
          style={
            Object {
              "opacity": 0.5,
            }
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`DiffSideBySide when setting the diff opacity to 0 1`] = `
<Fragment>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
  >
    <div
      className="viewer-received"
    >
      <img
        src="reports/something/received.png"
      />
      <div
        className="viewer-diff"
      >
        <img
          src="reports/something/diff.png"
          style={
            Object {
              "opacity": 0,
            }
          }
        />
      </div>
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
      <div
        className="viewer-diff"
      >
        <img
          src="reports/something/diff.png"
          style={
            Object {
              "opacity": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`DiffSideBySide when setting the diff opacity to 0.7 1`] = `
<Fragment>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={0.7}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
  >
    <div
      className="viewer-received"
    >
      <img
        src="reports/something/received.png"
      />
      <div
        className="viewer-diff"
      >
        <img
          src="reports/something/diff.png"
          style={
            Object {
              "opacity": 0.7,
            }
          }
        />
      </div>
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
      <div
        className="viewer-diff"
      >
        <img
          src="reports/something/diff.png"
          style={
            Object {
              "opacity": 0.7,
            }
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`DiffSideBySide when setting the diff opacity to 1 1`] = `
<Fragment>
  <nav
    className="level settings"
  >
    <div
      className="level-item has-text-centered"
    >
      <div>
        <div
          className="heading"
        >
          Diff Opacity
        </div>
        <input
          className="title"
          max={1}
          min={0}
          onChange={[Function]}
          step="any"
          type="range"
          value={1}
        />
      </div>
    </div>
  </nav>
  <div
    className="viewer-container"
  >
    <div
      className="viewer-received"
    >
      <img
        src="reports/something/received.png"
      />
      <div
        className="viewer-diff"
      >
        <img
          src="reports/something/diff.png"
          style={
            Object {
              "opacity": 1,
            }
          }
        />
      </div>
    </div>
    <div
      className="viewer-snapshot"
    >
      <img
        src="reports/something/snapshot.png"
      />
      <div
        className="viewer-diff"
      >
        <img
          src="reports/something/diff.png"
          style={
            Object {
              "opacity": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</Fragment>
`;
